### PR TITLE
Beamline fixes 4

### PR DESCRIPTION
**Pull request by Elisis**
# Additions
- Adds hot super coolant fluid for Synchrotron and LINAC coolant output, can be cooled back to regular super coolant in VF
- Add Prepared Masks for CPU, NCPU, PIC, HPIC. CR mask-to-mask recipes are replaced with mask-to-prepared mask, and prepared mask-to-mask engraver recipes are added. E.g. CPU-(CR)->PrCPU-(PLE)->NCPU.
- Adds beamline recipes for crystal chips, along with associated masks.

# Balancing
- Reduces UHPIC minimum TC energy to 190eV.
- Adds subticking to TC.
- TC recipe time save per increase in wafer tier improved by ~20%
- Synchrotron allowed to use laser/multiamp energy hatches.

# Fixes
- Fixes LINAC energy handling, now takes full amperage with two energy hatches.
- Fixes TC IOOB exception caused by empty focus input bus.
- Makes NPIC max energy sane.
- Increases LINAC coolant usage, now equal to length kL/s.
- Fixes OBOE with focus durability vs. max uses.
- Fixes equal amount values across different wafer tiers in TC recipes.
- Fixes tricoder NPEing on all multis that don't yet have beamline input hatches.
- Fixes empty Reactive-Ion Etching Mixture centrifuge recipe

# QoL
- Adds max number of uses of foci to their tooltips.
- Casing and item textures updated (Credit goes to BlueHero233)

### CODE REVIEW

**Comment by Dream-Master** [comment]
@boubou19 if it contain more fixes than changes I would add it to 2.7 . Thoughts ?

**Comment by FireWarrior2** [comment]
@Elisis this one also will be nice if fixed https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/17710

**Comment by Ruling-0** [comment]
> @boubou19 if it contain more fixes than changes I would add it to 2.7 . Thoughts ?

This really should get in - these are some pretty important fixes and balance changes.

**Comment by Elisis** [comment]
> @Elisis this one also will be nice if fixed [GTNewHorizons/GT-New-Horizons-Modpack#17710](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/17710)

Ah, that was the IOOB I mentioned

File: src/main/java/gtnhlanth/common/tileentity/MTESynchrotron.java [hunk 1]
```diff
@@ -620,19 +629,64 @@ public boolean addEnergyInputToMachineList(IGregTechTileEntity aTileEntity, int
...
+                    return mExoticEnergyHatches.add(hatchExotic);
+
+                } else {
+                    return false; // Not an energy hatch
+                }
+        } else {
+            return false; // Not a hatch of any kind
         }
-        return false;
+
```
Comment (Ethryan) [inline src/main/java/gtnhlanth/common/tileentity/MTESynchrotron.java:677]: Here

File: src/main/java/gtnhlanth/common/tileentity/recipe/beamline/BeamlineRecipeLoader.java [hunk 2]
```diff
@@ -165,6 +188,36 @@ public static void load() {
...
+            1,
+            24,
+            5,
+            12,
+            60,
+            1,
+            (int) TierEU.RECIPE_LuV);
+
+        /*
+         * BeamlineRecipeAdder2.instance.addTargetChamberRecipe( // TODO Test recipe for subticking
```
Comment (Ethryan) [inline src/main/java/gtnhlanth/common/tileentity/recipe/beamline/BeamlineRecipeLoader.java:208]: And Here

**Review by Ethryan — CHANGES_REQUESTED** [review]
Should the debug code still be there?, Might want to remove it if this is to be merged.

It also contains a few new items, but that's fine if its for balancing stuff imo.

PS: I just did a quick glance of this, and did no further testing.

**Review by Ethryan — APPROVED** [review]
Looks good imo.

**Pull request merged**